File: src/main/java/me/zeroeightsix/kami/module/modules/player/PitchLock.java
package me.zeroeightsix.kami.module.modules.player;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;
import net.minecraft.util.math.MathHelper;

/**
 * Created by 086 on 16/12/2017.
 */
@Module.Info(name = "PitchLock", category = Module.Category.PLAYER, description = "Locks your camera yaw")
public class PitchLock extends Module {
    private Setting<Boolean> auto = register(Settings.b("Auto", true));
    private Setting<Float> pitch = register(Settings.f("Pitch", 180));
    private Setting<Integer> slice = register(Settings.i("Slice", 8));

    @Override
    public void onUpdate() {
        if (slice.getValue() == 0) return;
        if (auto.getValue()) {
            int angle = 360 / slice.getValue();
            float yaw = mc.player.rotationPitch;
            yaw = Math.round(yaw / angle) * angle;
            mc.player.rotationPitch = yaw;
            if (mc.player.isRiding()) mc.player.getRidingEntity().rotationPitch = yaw;
        } else {
            mc.player.rotationPitch = MathHelper.clamp(pitch.getValue() - 180, -180, 180);
        }
    }
}

File: src/main/java/me/zeroeightsix/kami/module/modules/player/PitchLock.kt
package me.zeroeightsix.kami.module.modules.player

import me.zeroeightsix.kami.event.events.SafeTickEvent
import me.zeroeightsix.kami.module.Module
import me.zeroeightsix.kami.setting.Settings
import me.zeroeightsix.kami.util.event.listener
import kotlin.math.round

@Module.Info(
        name = "PitchLock",
        category = Module.Category.PLAYER,
        description = "Locks your camera pitch"
)
object PitchLock : Module() {
    private val auto = register(Settings.b("Auto", true))
    private val pitch = register(Settings.floatBuilder("Pitch").withValue(180.0f).withRange(-90.0f, 90.0f).withStep(1.0f))
    private val slice = register(Settings.integerBuilder("Slice").withValue(8).withRange(2, 32).withStep(1).withVisibility { auto.value })

    init {
        listener<SafeTickEvent> {
            if (auto.value) {
                val angle = 360.0f / slice.value
                mc.player.rotationPitch = round(mc.player.rotationPitch / angle) * angle
            } else {
                mc.player.rotationPitch = pitch.value
            }
        }
    }
}